### PR TITLE
use default tagging

### PR DESCRIPTION
**Pull request by anniehedgpeth**
## Background

A [task](https://app.asana.com/0/1181500399442529/1200343533684843/f) was done previously that standardizes on tagging across tests (and other repo modules), but the `OkToDelete` tag was neglected. Also, the change only affected the `common_tags` variable; it did not ensure that every taggable resource was being tagged. This is being done in this branch for the test suites by adding the `default_tags` to the AWS provider for the test modules while removing common tags on individual resources. Consequently, it required that I add a separate tag variable for the autoscaling group tags, as explained in the variable description.

[Asana Task](https://app.asana.com/0/1181500399442529/1200460471392854/f)

## How Has This Been Tested

`terraform plan` was run locally for each test. I will `/test all` here in the PR.

## This PR makes me feel

default

### CODE REVIEW

**Comment by anniehedgpeth** [comment]
That was a bad choice of gifs, and I am sorry if it gives you a headache. 😳

File: tests/private-active-active/main.tf [hunk 1]
```diff
@@ -56,7 +60,7 @@ module "private_active_active" {
   tfe_license_filepath        = ""
   tfe_subdomain               = "${local.friendly_name_prefix}-test-private-active-active"
 
-  common_tags = local.common_tags
+  asg_tags = local.common_tags
 }
 
 data "aws_instances" "main" {
```
Comment (aaron-lane) [inline tests/private-active-active/main.tf:66]: The filter logic for this data source is now insufficient as the internal proxy instance has the tags but not the key needed to authorize our SOCKS5 proxy connection. This isn't an issue for private-tcp-active-active as it was already designed to use the internal proxy instance for the SOCKS5 proxy as well: https://github.com/hashicorp/terraform-aws-terraform-enterprise/blob/annie-use-default-tagging/tests/private-tcp-active-active/outputs.tf#L34. We should update this case to do the same (and probably update the output names on both tests to be `proxy_instance_id`.

**Review by aaron-lane — CHANGES_REQUESTED** [review]
I identified the source of the latest test failure 👁️‍🗨️

**Review by aaron-lane — APPROVED** [review]
LGTM 🐛 

We'll take care of the latest private-active-active issue separately.

**Pull request merged**